hydrolib/core/basemodel.py

```diff
@@ -422,7 +446,10 @@ def retrieve_model(self, path: Path) -> Optional["FileModel"]:
                 The FileModel associated with the Path if it has been registered
                 before, otherwise None.
         """
-        return self._cache_dict.get(path, None)
+        file_model = self._cache_dict.get(path, None)
+        if file_model is None:
```

Comment:
These two lines are redundant. If the given `path` is not in the cache, `file_model` is `None` and will be returned on line 452.

Reply:
no, since line 452 is calling the model from the cache, the CachedFileModel, will rename it to cached_file_model